### PR TITLE
Refactor API and Workflows classes

### PR DESCRIPTION
**Pull request by Raruto**
## List of changes

- delete file `/api/index.js` (replaced by: `services/editingservice.js`)
- delete file `/toolboxes/toolboxesfactory.js` (replaced by: `services/toolboxes.js`)
- delete file `/toolboxes/toolsfactory.js` (replaced by: `services/toolboxes.js`)
- delete folder `/workflows/steps` (temporary replaced by: `g3wsdk/workflow/workflow.js`)
- delete folder `/workflows/steps/tasks` (temporary replaced by: `g3wsdk/workflow/workflow.js`)
- removal of all workflows child classes (replaced by base class: `EditingWorkflow`)
- extract some pure functions into `/utils` folder (mainly from base `EditingTask` class)

### CODE REVIEW

File: workflows/index.js
Comment (volterra79) [inline workflows/index.js:1]: @Raruto it could be useful to have multi files instead a big file with more than 2000 lines. It will be more maintainable

**Comment by Raruto** [comment]
@volterra79 Just to know, are they used by anyone else?

- [`g3w-client/src/app/core/workflow/flow.js`](https://github.com/g3w-suite/g3w-client/tree/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/flow.js)
- [`g3w-client/src/app/core/workflow/queque.js`](https://github.com/g3w-suite/g3w-client/tree/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/queque.js)
- [`g3w-client/src/app/core/workflow/step.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/step.js)
- [`g3w-client/src/app/core/workflow/task.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/task.js)
- [`g3w-client/src/app/core/workflow/workflow.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/workflow.js)

If not, let's start moving them into the editing plugin as soon as possible..

**Comment by volterra79** [comment]
> @volterra79 Just to know, are they used by anyone else?
> 
> * [`g3w-client/src/app/core/workflow/flow.js`](https://github.com/g3w-suite/g3w-client/tree/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/flow.js)
> * [`g3w-client/src/app/core/workflow/queque.js`](https://github.com/g3w-suite/g3w-client/tree/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/queque.js)
> * [`g3w-client/src/app/core/workflow/step.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/step.js)
> * [`g3w-client/src/app/core/workflow/task.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/task.js)
> * [`g3w-client/src/app/core/workflow/workflow.js`](https://github.com/g3w-suite/g3w-client/blob/54ad5a5c277f605c4a24b602ddf2e06d89f27f03/src/app/core/workflow/workflow.js)
> 
> If not, let's start moving them into the editing plugin as soon as possible..

@Raruto i checked. It seems no

**Review by volterra79 — APPROVED** [review]

File: workflows/index.js
Comment (Raruto) [reply workflows/index.js:1]: @volterra79 yes, I know, but let's take it into consideration in a future PR. (there are already too many changes in here..)

**Pull request merged**